applications/luci-app-usteer/htdocs/luci-static/resources/view/usteer/status.js

```diff
@@ -0,0 +1,264 @@
+'use strict';
+'require view';
+'require rpc';
+'require poll';
+'require dom';
+'require ui';
+
+return view.extend({
+	
+	callHostHints: rpc.declare({
+		object: 'luci-rpc',
+		method: 'getHostHints',
+		expect: { '': {} }
+	}),
+	callGetRemotehosts: rpc.declare({
+		object: 'usteer',
+		method: 'remote_hosts',
+		expect: {  '': {}}
+	}),
+	callGetRemoteinfo: rpc.declare({
+		object: 'usteer',
+		method: 'remote_info',
+		expect: { '': {} }
+	}),
+	callGetLocalinfo: rpc.declare({
+		object: 'usteer',
+		method: 'local_info',
+		expect: { '': {} }
+	}),
+	callGetClients: rpc.declare({
+		object: 'usteer',
+		method: 'get_clients',
+		expect: { '': {} }
+	}),
+
+
+	load: function() {
+		return Promise.all([
+			this.callHostHints(),
+			this.callGetRemotehosts(),
+			this.callGetRemoteinfo(),
+			this.callGetLocalinfo(),
+			this.callGetClients()
+			
+		]);
+	},
+	handleReset: null,
+	handleSaveApply: null,
+	handleSave: null,
+	
+	render: function(data) {
+	
+		var hosts = data[0];
+		var Remotehosts=data[1];
+		var Remoteinfo=data[2];
+		var Localinfo=data[3];
+		var Clients=data[4];
+
+
+
+		var body = E([
+			E('h2', _('Usteer status'))
+		]);
+
+
+/////////////////////////
+		body.appendChild(
+			E('h3', 'Remotehosts')
+		);
+		var remotehost_table = E('table', { 'class': 'table cbi-section-table' }, [
+			E('tr', { 'class': 'tr table-titles' }, [
+				E('th', { 'class': 'th' }, _('IP address')),
+				E('th', { 'class': 'th' }, _('identifier'))
+			])
+		]);
+		
+		var remotehosttableentries =[];
+		for(var IPaddr in Remotehosts) {
+			remotehosttableentries.push([
+							IPaddr, Remotehosts[IPaddr]['id']
+						]);
+				}
+
+		cbi_update_table(remotehost_table, remotehosttableentries, E('em', _('No data')));
+		body.appendChild(remotehost_table);
+		
+/////////////////////////
+		body.appendChild(
+			E('h3', 'Client list')
+		);
+		var connectioninfo_table = E('table', { 'class': 'table cbi-section-table' }, [
+			E('tr', { 'class': 'tr table-titles' }, [
+				E('th', { 'class': 'th' }, _('wlan')),
+				E('th', { 'class': 'th' }, _('bssid')),
+				E('th', { 'class': 'th' }, _('ssid')),
+				E('th', { 'class': 'th' }, _('freq')),
+				E('th', { 'class': 'th' }, _('n assoc')),
+				E('th', { 'class': 'th' }, _('noise')),
+				E('th', { 'class': 'th' }, _('load')),
+				E('th', { 'class': 'th' }, _('max assoc')),
+				E('th', { 'class': 'th' }, _('roam src')),
+				E('th', { 'class': 'th' }, _('roam tgt')),
+				E('th', { 'class': 'th' }, _('rrm_nr mac')),
+				E('th', { 'class': 'th' }, _('rrm_nr hex'))
+				
+			])
+		]);
+	
+		var connectioninfo_table_entries =[];
+		for(var wlan in Localinfo) {
+			connectioninfo_table_entries.push([
+							'<nobr>'+wlan+'</nobr>', 
```

Comment:
Is `wlan` the network SSID? If so, please change the code into `'<nobr>%h</nobr>'.format(wlan),` to HTML escape the value in order to avoid XSS.

Reply:
No, its the IP+interface name so something like "192.168.64.1#hostapd.phy0-ap0". Note this table only contains information about the users own network, so XSS should not be an issue. I can change it to that suggestion, but I do not think it is really needed. Let me know what you prefer